pkg/skaffold/build/gcb/cloud_build.go

```diff
@@ -92,8 +83,10 @@ func (b *Builder) buildArtifact(ctx context.Context, out io.Writer, tagger tag.T
 		return "", errors.Wrap(err, "uploading source tarball")
 	}
 
-	args := append([]string{"build", "--tag", artifact.ImageName, "-f", artifact.DockerArtifact.DockerfilePath}, buildArgs...)
+	args := append([]string{"build", "--tag", artifact.ImageName, "-f", artifact.DockerArtifact.DockerfilePath})
+	args = append(args, docker.GetBuildArgs(artifact.DockerArtifact)...)
```

Comment:
nice! so this is basically a bugfix besides being a dedupe, right? `docker.GetBuildArgs` handles the "single" build args, while this code above was assuming all buildargs were key-value type. Can we cover this with a testcase? :)